src/applications/coronavirus-research/config/submitForm.js

```diff
@@ -0,0 +1,64 @@
+import { submitToUrl } from 'platform/forms-system/src/js/actions';
+
+const submitForm = (form, formConfig) => {
+  const body = formConfig.transformForSubmit(formConfig, form);
+
+  const formData = form.data;
+  const uiSchema = formConfig.chapters.chapter1.pages.page1.uiSchema;
+
+  const getFormattedTrueSelectValues = element => {
+    return Object.keys(formData[element])
+      .filter(key => formData[element][key] === true)
+      .map(value => {
+        if (element === 'GENDER' && value.includes('SELF_IDENTIFY')) {
+          return `${uiSchema[element][value]['ui:title']}: ${
+            formData.GENDER_SELF_IDENTIFY_DETAILS
+          }`;
+        } else {
+          return uiSchema[element][value]['ui:title'];
+        }
+      });
+  };
+
+  const getFormattedRadioValues = (element, key) => {
+    return uiSchema[element]['ui:options'].labels[key];
+  };
+
+  const resolveBooleanValue = value => {
+    return value ? 'Yes' : 'No';
+  };
+
+  const eventData = {
+    'previously-diagnosed-with-covid19': formData.diagnosed ? 'Yes' : 'No',
```

Comment:
I'd love to normalize these around the JSON schema as much as possible (e.g. `previously-diagnosed-with-covid19` --> `diagnosed`. I know that doesn't follow the pattern of other GA event labels...but not sure if there is a requirement to use "dash case". Also maybe it is better to have more expressive event labels for the purposes of dashboard making. @stephenBarrs what says you?

Reply:
On second thought, I think the more expressive labels will be helpful, particular for our future selves who forget what's what. Please leave as is!